front/src/applications/stdcm/hooks/useStdcm.ts

```diff
@@ -87,74 +92,205 @@ const useStdcm = ({
     }
   };
 
+  const getPayloadWithTolerance = (payload: PostTimetableByIdStdcmApiArg, tolerance: number) => ({
+    ...payload,
+    body: {
+      ...payload.body,
+      steps: payload.body.steps.map((step) => modifyStepTolerance(step, tolerance)),
+    },
+  });
+
+  const createSimulation = async (
+    index: number,
+    inputs: StdcmSimulationInputs,
+    payload: PostTimetableByIdStdcmApiArg,
+    response: Extract<PostTimetableByIdStdcmApiResponse, { status: 'success' | 'conflicts' }>,
+    alternativePath?: 'upstream' | 'downstream'
+  ): Promise<StdcmSimulation> => {
+    const formattedResponse = {
+      ...response,
+      rollingStock: stdcmRollingStock,
+      creationDate: new Date(),
+      speedLimitByTag: osrdconf.speedLimitByTag,
+      simulationPathSteps: osrdconf.stdcmPathSteps,
+      path: response.status === 'conflicts' ? response.pathfinding_result : response.path,
+    } as StdcmResponse;
+
+    const pathProperties = await fetchPathProperties(formattedResponse.path, infraId, dispatch);
+    if (!pathProperties) {
+      throw new Error('Path properties could not be fetched.');
+    }
+
+    // If the response is successful compute the chart data, otherwise only include conflicts.
+    let outputs;
+    if (formattedResponse.status === 'success') {
+      const stdcmTrain: TrainScheduleResult = {
+        id: STDCM_TRAIN_ID,
+        timetable_id: timetableId,
+        comfort: payload.body.comfort,
+        constraint_distribution: 'MARECO',
+        path: payload.body.steps.map((step) => ({ ...step.location, id: nextId() })),
+        rolling_stock_name: stdcmRollingStock!.name,
+        start_time: formattedResponse.departure_time,
+        train_name: 'stdcm',
+      };
+      const chartData = computeChartData(
+        formattedResponse,
+        stdcmTrain,
+        t,
+        stdcmRollingStock as RollingStockWithLiveries,
+        pathProperties
+      );
+      outputs = {
+        pathProperties,
+        results: formattedResponse,
+        speedSpaceChartData: chartData,
+      };
+    } else {
+      outputs = { pathProperties, conflicts: formattedResponse.conflicts ?? [] };
+    }
+
+    return {
+      index,
+      creationDate: formattedResponse.creationDate,
+      inputs,
+      outputs,
+      alternativePath,
+    };
+  };
+
+  const handleSuccess = async (
+    response: StdcmApiSuccessResponse,
+    payload: PostTimetableByIdStdcmApiArg
+  ) => {
+    setCurrentStdcmRequestStatus(STDCM_REQUEST_STATUS.success);
+    dispatch(updateSelectedTrainId(STDCM_TRAIN_TIMETABLE_ID));
+
+    const simulation = await createSimulation(
+      simulationsList.length,
+      currentSimulationInputs,
+      payload,
+      response
+    );
+    dispatch(addStdcmSimulation(simulation));
+    dispatch(setSelectedSimulationIndex(simulation.index));
+  };
+
+  const handleConflicts = async (
+    response: StdcmApiConflictsResponse,
```

Comment:
Same here, you can remove StdcmApiConflictsResponse
```suggestion
    response: Extract<PostTimetableByIdStdcmApiResponse, { status: 'conflicts' }>,
```